src/main/kotlin/com/expedia/graphql/KotlinDataFetcher.kt

```diff
@@ -39,11 +47,27 @@ class KotlinDataFetcher(
                 if (annotations.any { it.annotationClass == GraphQLContext::class }) {
                     environment.getContext()
                 } else {
-                    mapper.convertValue(environment.arguments[name], klazz)
+                    parseArgument(environment, name, klazz, annotations)
                 }
             }.toTypedArray())
         } else {
             null
         }
+
+    }
+
+    private fun parseArgument(environment: DataFetchingEnvironment, name: String, klazz: Class<*>, annotations: List<Annotation>): Any? {
```

Comment:
this will attempt to parse annotations for each query/field invocation which might be an overkill